internal/cmd/main.go

```diff
@@ -1,13 +1,79 @@
-// Package main defines the entry point for the program.
 package main
 
-import "log"
+import (
+	"context"
+	"flag"
+	"os"
+	"os/signal"
+	"syscall"
 
-// Greet returns a greeting message.
-func Greet() string {
-	return "Hello, world!"
-}
+	"github.com/ezex-io/ezex-users/internal/core/common/router"
+	"github.com/ezex-io/ezex-users/internal/core/server"
+	"github.com/ezex-io/ezex-users/internal/core/service"
+	"github.com/ezex-io/ezex-users/internal/infra/config"
+	"github.com/ezex-io/ezex-users/internal/infra/logger"
+	"github.com/ezex-io/ezex-users/internal/infra/repository"
+	"github.com/joho/godotenv"
+)
 
 func main() {
-	log.Println(Greet())
+	envFile := flag.String("env", ".env.example", "Path to environment file")
+	flag.Parse()
+
+	logging := logger.NewSlog(nil)
+
+	if err := godotenv.Load(*envFile); err != nil {
+		logging.Warn("Failed to load env file '%s': %v. Continuing with system environment...", *envFile, err)
+	} else {
+		logging.Debug("Loaded environment variables from '%s'", *envFile)
+	}
+
+	cfg, err := config.Load()
+	if err != nil {
+		logging.Fatal("Failed to load config: %v", err)
+	}
+
+	if err := cfg.BasicCheck(); err != nil {
+		logging.Fatal("Configuration validation failed: %v", err)
+	}
+
+	httpRouter := router.SetupRouter()
+	httpServer := server.NewHTTPServer(cfg.HTTPServer.Address, httpRouter)
+
+	securityImageService := service.NewSecurityImageService(
+		repository.NewSecurityImageRepository(),
+	)
+	grpcServer := server.NewGRPCServer(cfg.GRPCServer.Address, securityImageService)
+
+	go func() {
+		logging.Info("Starting HTTP server on %s", cfg.HTTPServer.Address)
+		if err := httpServer.Start(); err != nil {
```

Comment:
Please remove the HTTP service. It is a plugin over gRPC and can be managed externally without injecting any code into the original service.